dask/dataframe/core.py

```diff
@@ -5055,7 +5058,20 @@ def eval(self, expr, inplace=None, **kwargs):
         return self.map_partitions(M.eval, expr, meta=meta, inplace=inplace, **kwargs)
 
     @derived_from(pd.DataFrame)
-    def dropna(self, how="any", subset=None, thresh=None):
+    def dropna(self, how=pd_no_default, subset=None, thresh=pd_no_default):
+
+        if (
+            not PANDAS_GT_150
+            and (how is not pd_no_default)
+            and (thresh is not pd_no_default)
+        ):
+            raise ValueError(
```

Comment:
Pandas error is a TypeError, but when we catch the errors in here https://github.com/dask/dask/blob/f7fd6c480478bdee5c7d6b8736ba69d5a2e918e0/dask/dataframe/utils.py#L215-L216 we use a ValueError so I'm using the same to be consistent.

Reply:
Interesting, I wonder why they raise a `TypeError`, `ValueError` seems more semantically correct to me as well.

I *might* prefer to be more consistent with pandas, so that user code could theoretically catch the same exception whether using dask or pandas, but I am not sure it's a huge deal either way

Reply:
Yeah, I had the same thought and preference as you. I'll go ahead and make this a `TypeError`. Like you said, probably not a big deal either way, but could lead to unexpected behavior if coming from `pandas`